### ISSUE TITLE
FluentAssertions extensions do not filter headers correctly

### ISSUE THREAD

**Issue by chrischu** [WireMock-Net/WireMock.Net#1074]
### Describe the bug
When asserting headers via the FluentAssertions extensions, headers are not filtered correctly (headers from all captured requests are checked).

### Expected behavior:
I expect the assertions to only consider the requests that match the previous filters (see example below).

### Test to reproduce
``` c#
[Test]
public async Task SendGetWithHeader ()
{
  using var server = WireMockServer.Start();
  using var httpClient = new HttpClient();

  // ACT
  await httpClient.GetAsync(server.Url!);

  var request = new HttpRequestMessage(HttpMethod.Post, server.Url!);
  request.Headers.Add("TestHeader", "Value");
  await httpClient.SendAsync(request);

  // ASSERT
  server.Should().HaveReceivedACall().UsingGet().And.WithHeader("TestHeader", "Value");
}
```

This test succeeds although it should fail, since only the POST request has the header, the GET request does not.

### Other related info
I'm using WireMock.Net & WireMock.Net.FluentAssertions 1.5.48.

**Comment by StefH** [comment]
@chrischu 
I can confirm this bug and it's fixed in preview version 1.5.49-ci-18411

**Comment by chrischu** [comment]
Thanks for the quick fix!
